File: Behavioral/Lab19/Typescript/src/Entities/customer.ts
import { IBuyable } from '../Interfaces/buyable';
import { Person } from './person';

export class Customer extends Person {
    constructor(nameAndSurname: string, phone: string, email: string);
    constructor();
    constructor(nameAndSurname?: string, phone?: string, email?: string) {
        if (nameAndSurname && phone && email) {
            super(nameAndSurname, phone, email);
        } else {
            super();
        }
    }

    public setToBuy(buyable: IBuyable, price: number): void {
        buyable.setEstateToBuy(this, price);
    }

    public notify(message: string): void {
        console.log(`${this.nameAndSurname} was notified about ${message}`);
    }
}

File: Behavioral/Lab16/C#/Entities/Customer.cs
using C_.Interfaces;

namespace C_.Entities
{
    public class Customer : Person
    {
        public Customer(string nameAndSurname, string phone, string email):base(nameAndSurname, phone, email) { }

        public Customer():base(){}

        public void SetToBuy(IBuyable buyable, double price) {
            buyable.SetEstateToBuy(this, price);
        }
    }

}

